File: LeetCode/Easy/2899-last-visited-integers/2899-last-visited-integers.py
class Solution:
    def lastVisitedIntegers(self, nums: List[int]) -> List[int]:
        seen = deque()
        ret = []
        k = 0
        
        for v in nums:
            if v != -1:
                k = 0
                seen.appendleft(v)
            else:
                k += 1

                if k > len(seen):
                    ret.append(-1)
                else:
                    ret.append(seen[k-1])
                
        return ret

File: LeetCode/Easy/2899-last-visited-integers/2899-last-visited-integers.java
class Solution {
    public List<Integer> lastVisitedIntegers(int[] nums) {
        LinkedList<Integer> seen = new LinkedList();
        List<Integer> ret = new ArrayList();
        int k = 0;
        
        for (int v : nums){
            if (v != -1){
                seen.addFirst(v);
                k = 0;
            } else {
                ++k;
                
                if (k > seen.size()) ret.add(-1);
                else ret.add(seen.get(k-1));
            }
        }
        
        return ret;
    }
}